### DISCUSSION TITLE
Custom Anchor Names During Serilialize?

### DISCUSSION THREAD

**Discussion by BBoldenow-Alt** [aaubry/YamlDotNet#744] [Q&A] [answered]
Is there a way to customize the name of an anchor during serialization? I can easily output a custom AnchorAlias using a type converter (e.g. "*custom_anchor_name"), but not the anchor for the originating object (e.g. "&custom_anchor_name"). None of the ways I've come up with of doing it seem to output it properly, and I was hoping to have names that were easier to reference than the default generated ones.

**Comment by Nidhi1819** [comment]
I am having the same issue and looking for a resolution. Did you find any way to get this done.

**Reply by BBoldenow-Alt** [reply]
No, I wrote my own serializer using a TypeConverter, but when I emitted a different anchor name, it would re-serialize the entire object with a default anchor name when serializing an object with a reference to that original object. 

So it would look like this (or something very similar - this is off the top of my head):

```
- &my_instance !CustomClassOne
  InstanceName: my_instance
- !CustomClassTwo
  Context: &o0 !CustomClassOne
      InstanceName: my_instance
  ParameterName: some_value
- !CustomClassTwo
  Context: *o0
  ParameterName: some_value
```

My guess is I'm going to have to customize this library to do it myself, but it hasn't been a high enough priority to this point...I'll get to it eventually.

**Reply by Nidhi1819** [reply]
Thanks @BBoldenow-Alt , hope there was better way to do this rather than customizing the library. I tried implementing IYamlTypeConverter  and it did generate custom names for anchor tag but it did **not** generate alias for repeated objects. Looks like will have to override the complete behavior to achieve what I want but since I do not know how library works it is difficult to change.

**Comment by BBoldenow-Alt** [comment]
That sounds like what I saw. When I have time the way I was planning to
implement it is to create a custom attribute the library would read called
[YamlAnchor].

Good luck, sorry I haven’t gotten around to it myself yet!

**Comment by EdwardCooke** [comment, marked as answer]
Take a look at my suggestion #807

**Reply by BBoldenow-Alt** [reply]
@Nidhi1819 

You may have already seen this since it was your question, but take a look at this suggestion, and the additional context I provided after it was closed. This works for me now. Custom anchor names using data from the object, exactly like I was hoping to do via a custom attribute. This was a lot faster though.
